### ISSUE TITLE
Volume metrics doesn't update after volume migration

### ISSUE THREAD

**Issue by choleowsky** [apache/cloudstack#3878]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME

~~~
CloudRuntimeException
~~~

##### CLOUDSTACK VERSION

~~~
4.11.2.0
~~~

##### CONFIGURATION

Advanced network
Qemu-KVM hypervisor
NFS shared storage
##### OS / ENVIRONMENT

Ubuntu 16.04 / 18.04

##### SUMMARY

Error in CloudRuntimeException: Can't find volume:
##### STEPS TO REPRODUCE

* Install fresh cloudstack 4.11.2.0 whith Qemu-KVM hosts, NFS primary storage server and advanced network
* Create two new instances
* Add new primary storage
* Migrate one instance on new primary storage server
~~~
~~~

##### EXPECTED RESULTS

Volume metrics are updated on all sotrage server
~~~

~~~

##### ACTUAL RESULTS

The migration works but the volume metrics doesn't update in new storage server, on the first one are updated as aspected.

This error appear in agent log:

~~~
2020-01-14 09:10:47,419 WARN  [cloud.agent.Agent] (agentRequest-Handler-3:null) (logid:6d27b740) Caught:
com.cloud.utils.exception.CloudRuntimeException: Can't find volume:d93d3c0a-3859-4473-951d-9b5c5912c767
at com.cloud.hypervisor.kvm.storage.LibvirtStoragePool.getPhysicalDisk(LibvirtStoragePool.java:149)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetVolumeStatsCommandWrapper.getVolumeStat(LibvirtGetVolumeStatsCommandWrapper.java:63)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetVolumeStatsCommandWrapper.execute(LibvirtGetVolumeStatsCommandWrapper.java:52)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetVolumeStatsCommandWrapper.execute(LibvirtGetVolumeStatsCommandWrapper.java:40)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtRequestWrapper.execute(LibvirtRequestWrapper.java:78)
at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.executeRequest(LibvirtComputingResource.java:1450)
at com.cloud.agent.Agent.processRequest(Agent.java:645)
at com.cloud.agent.Agent$AgentRequestHandler.doTask(Agent.java:1083)
at com.cloud.utils.nio.Task.call(Task.java:83)
at com.cloud.utils.nio.Task.call(Task.java:29)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
2020-01-14 09:20:48,390 INFO  [kvm.storage.LibvirtStorageAdaptor] (agentRequest-Handler-4:null) (logid:ec72387b) Trying to fetch storage pool 9af0d1c6-85f2-3c55-94af-6ac17cb4024c from libvirt
2020-01-14 09:20:48,536 WARN  [cloud.agent.Agent] (agentRequest-Handler-4:null) (logid:ec72387b) Caught:
com.cloud.utils.exception.CloudRuntimeException: Can't find volume:d93d3c0a-3859-4473-951d-9b5c5912c767
at com.cloud.hypervisor.kvm.storage.LibvirtStoragePool.getPhysicalDisk(LibvirtStoragePool.java:149)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetVolumeStatsCommandWrapper.getVolumeStat(LibvirtGetVolumeStatsCommandWrapper.java:63)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetVolumeStatsCommandWrapper.execute(LibvirtGetVolumeStatsCommandWrapper.java:52)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtGetVolumeStatsCommandWrapper.execute(LibvirtGetVolumeStatsCommandWrapper.java:40)
at com.cloud.hypervisor.kvm.resource.wrapper.LibvirtRequestWrapper.execute(LibvirtRequestWrapper.java:78)
at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.executeRequest(LibvirtComputingResource.java:1450)
at com.cloud.agent.Agent.processRequest(Agent.java:645)
at com.cloud.agent.Agent$AgentRequestHandler.doTask(Agent.java:1083)
at com.cloud.utils.nio.Task.call(Task.java:83)
at com.cloud.utils.nio.Task.call(Task.java:29)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
~~~

**Comment by choleowsky** [comment]
I found a wrkaround that seems to indicate where the problem lies.

From what I have seen, the LibvirtStoragePool.getPhysicalDisk procedure goes to look for the files of the migrated volumes with the value of the "path" column of the "volumes" table.
When a volume is migrated from one storage to another, the "path" field is changed but the LibvirtStoragePool.getPhysicalDisk procedure seems to go looking for the volume with its old "path" which now corresponds to the "uuid" field.

If I make a symbolic link directly in the new NFS storage server's filesystem (ln -s <path> <uuid>), after some time I found that metrics were updated and the message in agent.log no longer appeared.
